clippy_lints/src/loops/manual_flatten.rs

```diff
@@ -54,20 +55,35 @@ pub(super) fn check<'tcx>(
             _ => "",
         };
 
-        let sugg = format!("{arg_snippet}{copied}.flatten()");
+        let help_msg = "try `.flatten()` and remove the `if let` statement in the for loop";
 
-        // If suggestion is not a one-liner, it won't be shown inline within the error message. In that
-        // case, it will be shown in the extra `help` message at the end, which is why the first
-        // `help_msg` needs to refer to the correct relative position of the suggestion.
-        let help_msg = if sugg.contains('\n') {
-            "remove the `if let` statement in the for loop and then..."
+        let pat_snippet = if let_pats.is_empty() {
+            "_".to_string()
         } else {
-            "...and remove the `if let` statement in the for loop"
+            snippet_with_applicability(
+                cx,
+                let_pats.first().unwrap().span.source_callsite(),
+                "_",
+                &mut applicability,
```

Comment:
Shouldn't you initialize `applicability` with `MachineApplicable` if the suggestion is complete?